### ISSUE TITLE
Request a clean update option (completely remove and replace mod, but keep load order)

### ISSUE THREAD

**Issue by lauren-aka-proudunicornmods** [Nexus-Mods/Vortex#8477]
**Problem:**

When updating mods that have changed their internal file structure or added additional files, Vortex's dialogue to save/revert changes actually takes a lot longer to use (if a lot of changes) than manually installing the mod. Consequently, many SV mods (including some of mine) choose to block Vortex downloads.

This is especially a problem in premium, where the update button does not even take you to the download page (where the mod author may have explained that it requires a clean install).

**Solution**: 

Either A) add a **'clean update'** button, which completely deletes the mods files, but keeps its entry (and therefore load order), and then downloads and installs the new mod in its entirety.

Or B) create an option for mod users to **'require manual update',** i.e. force Vortex to display the button in screenshot below:

**Screenshot**:
![image](https://user-images.githubusercontent.com/71786398/107235595-c021c800-6a1c-11eb-9a0c-673f46300ee0.png)

**Comment by IDCs** [comment]
Hi @lauren-aka-proudunicornmods thank you for reporting this to us. By SV I'm assuming you mean Stardew Valley. Vortex's automatic update system already offers a 'clean update' functionality when installing a new version of a mod through the below dialog - the "Update current profile" option will install the new mod version alongside the older version while "Update all profiles" will completely replace the older mod version with the new version.

![image](https://user-images.githubusercontent.com/8960252/107331639-12a9c580-6aab-11eb-8d28-51fa4d873f30.png)

Unfortunately, this dialog will only appear if the mod author had correctly highlighted the fact that version X of his mod supersedes version Y when he uploads the new version. The internal file structure of the mod would make no difference, as far as Vortex is concerned if the user selected to "Update all profiles", the older version of the mod is removed completely, and the newer version is installed in its place; if that is not what is happening on your end, and you've confirmed that the mod author had defined the version succession correctly - then that's indeed a bug which I'm unable to reproduce on my end... Please let us know if you're able to reproduce this reliably, and provide a step-by-step guide for us to try to debug this.

As far as I know, SMAPI loads the mods into the game in alphabetical order, which means that as long as the mod's name hasn't changed, the load order should remain the same, but if you mean the mod's position in the mods table inside Vortex - that will be maintained as long as the mod's value for the column you're sorting by remains unchanged. Regardless, Vortex itself plays no part in how SMAPI decides to order your mods - any load order issues should be forwarded to the SMAPI developer.

What concerns me in your report is the fact that you appear to have encountered the external changes dialog when updating a mod which should never happen! Vortex monitors the game's mods directory and should notice any changes that occur during game time and apply them to the mod in the staging folder (which should ensure that Vortex doesn't raise the EC dialog) but that obviously will not work if the mod user overwrites or changes files inside the game's mods folder manually.

Could you perhaps assist us in debugging the external changes dialog issue you're having and provide us with a step-by-step guide to reproducing this ? Please include the mod links and specify the exact versions of the mods you've used when reproducing it - also provide screenshots of the game's mods folder - a before/after sort of thing. We would also require your log files which capture this behaviour - they are located inside the "%appdata%/Vortex/" directory; please include all files with the ".log" extension.

In response to your solution suggestions:
A) already implemented and should work as long as Vortex is able to identify that version X supersedes version Y (this is done by the mod author when they upload a new version of the mod)

B) I don't see any reason why we couldn't include the icon button you displayed in your screenshot even when the ability to automatically update exists (for premium users) @TanninOne what do you think about this ?

**Comment by lauren-aka-proudunicornmods** [comment]
Thank you for such a quick and detailed reply!

To answer your questions:

1) I was not aware that this is how updates should work as it pretty much always creates issues if I 'update for all profiles' a mod which has changed internal structure / added removed files. 

2) I often get the 'external changes' message and I am happy to help with providing the information that you asked for. I have many mods, so I will try and find one that has changes to structure in a recent update and then install an old version and then update it :)

3) I would LOVE if you could make that button an option!

4) Also, FYI I am using the Symlink option as that is the default for Stardew. Maybe that is not the correct option? 

5) Additionally FYI: I recently uploaded a new copy of my mod here: https://www.nexusmods.com/stardewvalley/mods/6618?tab=files
and I did not mark it as a replacement version.
What I did was: 
a) Moved the main file to 'old files'
b) Uploaded a new version
c) didn't check 'mark as a new version of existing file'
d) didn't check 'make main vortex download'
e) changed the name of file
f) but I didn't uncheck 'allow users to download through vortex'
and the update option in Vortex still showed the cloud auto update. I was hoping it would show the manual update button. 

Thanks so much for your message. I will try to recreate the issue and screenshot it for you + provide the logs later :)

**Comment by IDCs** [comment]
No worries - we try to ensure that Vortex satisfies all modding requirements for the games we support, but at this point we support so many that it's hard for us to detect flaws in our design without our userbase reaching out to us, so we're grateful for your assistance in debugging this.

One important note - the External Changes Dialog (ECD) is a completely separate component from the mods update component. The ECD component monitors mod files that have been installed and deployed through Vortex, and aims to detect any external modifications committed to those files and allow users to either revert or save the change. Updating a mod should **never** trigger the ECD as regardless of how you install the update, the mods are self contained and any overlapping files would be reported as conflicts instead.

1. As I explained previously, any changes to the internal file structure of the mod do not matter - 'update for all profiles' should completely replace the older version of the mod with the new version and should never trigger the ECD - the fact that it does, suggests that your issue might be down to a well known flaw in our conflict resolution system where we're unable to identify conflicting mod files if the mods are of different modTypes - I should be able to confirm this if you can send us your deployment manifests:

- Can you please attach any .json file you can find in the game's root folder (e.g. "D:\Games\SDV\") the files we're interested in start with "vortex.deployment" and end with ".json" - you should at least have a "vortex.deployment.SMAPI.json"
- Also please attach the "vortex.deployment.json" file which is found inside the game's mods folder (e.g. "D:\Games\SDV\Mods")
(Simply drag and drop these files to your response to attach them)

4. Symlinks should work fine with SDV - that's what I use as well.
5. I've installed an older version of your mod and the "open mod page" icon is presented instead of the auto update cloud icon.
![image](https://user-images.githubusercontent.com/8960252/107368315-4c44f580-6ad8-11eb-9f6a-d080b0b31bd8.png)

**Comment by IDCs** [comment]
Hey @lauren-aka-proudunicornmods, just tagging you in case you didn't see my previous comment.

tl;dr:
* Can you please attach any .json file you can find in the game's root folder (e.g. "D:\Games\SDV") the files we're interested in start with "vortex.deployment" and end with ".json" - you should at least have a "vortex.deployment.SMAPI.json"
* Also please attach the "vortex.deployment.json" file which is found inside the game's mods folder (e.g. "D:\Games\SDV\Mods")
(Simply drag and drop these files to your response to attach them)

**Comment by IDCs** [comment]
Hi @lauren-aka-proudunicornmods, given your lack of activity I'm assuming you no longer need our assistance regarding the external changes issue, so I'm going to close this report.

After some thought regarding the manual update icon, I think the use case you describe and your reasoning behind wanting this functionality is invalidated by the fact that you misunderstood how Vortex is supposed to work + the fact that you are probably affected by a problematic design flaw (which we've yet to fix) where we cannot detect file conflicts between different modTypes, which is probably the root cause behind the external changes dialog being raised every time you deploy/purge (as two mods are repeatedly replacing each other's files). We have an open issue which we use to track that particular bug - https://github.com/Nexus-Mods/Vortex/issues/5869

As a side-note your mod's description and the ability to open the mod's page is always available via the mod's side panel which can be opened by double clicking any mod inside your mods table.
![image](https://user-images.githubusercontent.com/8960252/108474565-768c7500-7287-11eb-9cc5-df0d21941160.png)


duplicate of #5869
